plugins/base/src/test/kotlin/transformers/ReportUndocumentedTransformerTest.kt

```diff
@@ -0,0 +1,832 @@
+            passes {
+                pass {
+                    reportUndocumented = true
+                    sourceRoots = listOf("src/main/kotlin/Test.kt")
+                }
+            }
+        }
+
+        testInline(
+            """
+            |/src/main/kotlin/Test.kt    
+            |package sample
+            |
+            |internal class X
+            """.trimIndent(),
+            configuration
+        ) {
+            pagesTransformationStage = {
+                assertNoUndocumentedReport()
+            }
+        }
+    }
+
+    @Test
+    fun `undocumented function gets reported`() {
+        val configuration = dokkaConfiguration {
+            passes {
+                pass {
+                    reportUndocumented = true
+                    sourceRoots = listOf("src/main/kotlin/Test.kt")
+                }
+            }
+        }
+
+        testInline(
+            """
+            |/src/main/kotlin/Test.kt    
+            |package sample
+            |
+            |/** Documented */
+            |class X {
+            |    fun x()
+            |}
+            """.trimIndent(),
+            configuration
+        ) {
+            pagesTransformationStage = {
+                assertSingleUndocumentedReport(Regex("X"))
+                assertSingleUndocumentedReport(Regex("X/x"))
+            }
+        }
+    }
+
+    @Test
+    fun `undocumented property gets reported`() {
+        val configuration = dokkaConfiguration {
+            passes {
+                pass {
+                    reportUndocumented = true
+                    sourceRoots = listOf("src/main/kotlin/Test.kt")
+                }
+            }
+        }
+
+        testInline(
+            """
+            |/src/main/kotlin/Test.kt    
+            |package sample
+            |
+            |/** Documented */
+            |class X {
+            |    val x: Int = 0
+            |}
+            """.trimIndent(),
+            configuration
+        ) {
+            pagesTransformationStage = {
+                assertSingleUndocumentedReport(Regex("X"))
+                assertSingleUndocumentedReport(Regex("X/x"))
+            }
+        }
+    }
+
+    @Test
+    fun `undocumented primary constructor does not get reported`() {
+        val configuration = dokkaConfiguration {
+            passes {
+                pass {
+                    reportUndocumented = true
+                    sourceRoots = listOf("src/main/kotlin/Test.kt")
+                }
+            }
+        }
+
+        testInline(
+            """
+            |/src/main/kotlin/Test.kt    
+            |package sample
+            |
+            |/** Documented */
+            |class X(private val x: Int) {
+            |}
+            """.trimIndent(),
+            configuration
+        ) {
+            pagesTransformationStage = {
+                assertNoUndocumentedReport()
+            }
+        }
+    }
+
+    @Disabled
```

Comment:
I am not sure if a secondary constructor should *always* be documented by default 🤔